hackathon_site/dashboard/frontend/src/pages/Inventory/Inventory.test.tsx

```diff
@@ -1,14 +1,24 @@
 import React from "react";
 
-import { makeMockApiListResponse, render, waitFor, when } from "testing/utils";
+import {
+    makeMockApiListResponse,
+    render,
+    waitFor,
+    when,
+    fireEvent,
+} from "testing/utils";
 
 import Inventory from "pages/Inventory/Inventory";
 import { mockCategories, mockHardware } from "testing/mockData";
 
-import { get } from "api/api";
+import { get, stripHostnameReturnFilters } from "api/api";
 
 jest.mock("api/api");
 const mockedGet = get as jest.MockedFunction<typeof get>;
+const mockStripHostnameReturnFilters =
```

Comment:
Do we have to mock this? I think it would be better to just mock the API endpoints (via `get`), but leave this function as-is.

If you still want to mock this function, you should add an explicit unit test for it to be safe.

Reply:
yeah so I didn't think I needed to mock this either but the unit tests will fail if I don't :( I started logging a bunch of things to figure out why it was failing and the url is undefined after passing through that function 😔 I'm not sure why though but mocking it fixed that 🤷‍♀️

Reply:
That would make me think that there might be something wrong with the function if it is returning undefined, since that isn't even a valid return type for it. If it helps, set a breakpoint in there and then run the test in the pycharm debugger!

Reply:
Aha, I know why it's returning undefined! When you do `jest.mock("api/api")`, that actually mocks everything in that module (mocking is the only thing I really don't like about Jest, it's so much easier in Python).

I got around this in the [category slice tests](https://github.com/ieeeuoft/hackathon-template/blob/develop/hackathon_site/dashboard/frontend/src/slices/hardware/categorySlice.test.ts#L18) by mocking just the get function:

```javascript
jest.mock("api/api", () => ({
    ...jest.requireActual("api/api"),
    get: jest.fn(),
}));
```

Should work for you, and allow you to use the actual `stripHostnameReturnFilters`.